src/lib/protocol/arp.lua

```diff
@@ -0,0 +1,105 @@
+module(..., package.seeall)
+
+-- RFC 826 (ARP) requests and replies.
+-- Note: all incoming configurations are assumed to be in network byte order.
+
+local ethernet = require("lib.protocol.ethernet")
+local header = require("lib.protocol.header")
+local packet = require("core.packet")
+
+local ffi = require("ffi")
+local C = ffi.C
+
+--[[ Packet format (IPv4/Ethernet), as described on Wikipedia.
+
+Internet Protocol (IPv4) over Ethernet ARP packet
+octet offset 	0 	1
+0 	Hardware type (HTYPE)
+2 	Protocol type (PTYPE)
+4 	Hardware address length (HLEN) 	Protocol address length (PLEN)
+6 	Operation (OPER)
+8 	Sender hardware address (SHA) (first 2 bytes)
+10 	(next 2 bytes)
+12 	(last 2 bytes)
+14 	Sender protocol address (SPA) (first 2 bytes)
+16 	(last 2 bytes)
+18 	Target hardware address (THA) (first 2 bytes)
+20 	(next 2 bytes)
+22 	(last 2 bytes)
+24 	Target protocol address (TPA) (first 2 bytes)
+26 	(last 2 bytes)
+--]]
+
+local arp_header_t = ffi.typeof[[
+   struct {
+   uint16_t arp_htype;
+   uint16_t arp_ptype;
+   uint8_t arp_hlen;
+   uint8_t arp_plen;
+   uint16_t arp_oper;
+   uint8_t arp_sha[6];
+   uint8_t arp_spa[4];
+   uint8_t arp_tha[6];
+   uint8_t arp_tpa[4];
+   } __attribute__((packed))
+]]
+
+local arp = subClass(header)
+-- ARP class variables
+arp._name = "arp"
+arp._header_type = arp_header_t
+arp._header_ptr_type = ffi.typeof("$*", arp_header_t)
+arp._ulp = { method = nil }
+
+-- Constants
+local arp_request = C.ntohs(1)
```

Comment:
These are constants used to construct an ARP packet, if I'm correct. Wouldn't the cast be the opposite, from host byte order to network byte order?